### PR TITLE
user can choose cpu difficulty

### PR DESCRIPTION
**Pull request by Cristian-Baeza**

### CODE REVIEW

File: lib/game_loop.rb [hunk 1]
```diff
@@ -27,7 +27,11 @@ def core_game
         user_choice = @questions.validate_user_input("PICK A SPACE #{@game_logic.current_player()}", valid_options)
         @game_logic.take_turn(user_choice)
       else
-        @computer_logic.cpu_turn
+        if @game_config.cpu_difficulty == :easy
+          @computer_logic.cpu_turn_easy
+        else
+          @computer_logic.cpu_turn_hard
+        end
```
Comment (td-extreme) [inline lib/game_loop.rb:34]: There is no reason that the game loop should be modified for this change.

**Review by td-extreme — CHANGES_REQUESTED** [review]
Can you complete this without modifying the game_loop.

File: lib/game_loop.rb (hunk 1 above)
Comment (Cristian-Baeza) [reply lib/game_loop.rb]: Got it! It was a doozy adjusting test specs to pass but figured it out

**Comment by Cristian-Baeza** [comment]
> Can you complete this without modifying the game_loop?

Got a working solution that does not change game_loop. 
-Changed game_loop back to the way it was 
- Computer_logic now uses game_config to keep track of game difficulty
- Adjusted tests as needed for them to pass based on the changes made, mainly computer_logic now needing game_config during initialize

**Comment by td-extreme** [comment]
Per our stand up talk -
* Let's change the `ComputerLogic` class into `ComputerLogicHard` and create a new `ComputerLogicEasy` class and test file.
* Then have the main bin/tictactoe.rb script decide which class to pass to the game_loop
* While were at it can we put the computer players into their own folder to organize things a little better
`/lib/computer_logic/computer_logic_easy.rb`
`/lib/computer_logic/computer_logic_hard.rb`

**Review by td-extreme — APPROVED** [review]

**Pull request merged**